interfaces/inc/file.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef SYNC_H
+#define SYNC_H
+
+#ifdef __cplusplus
+#include <cstdbool>
+#else
+#include <stdbool.h>
+#endif
+
+#include "umock_c/umock_c_prod.h"
+#include "constbuffer.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define FILE_DATA_RESULT_VALUES \
+    FILE_DATA_FAILURE, \
+    FILE_DATA_INVALID, \
+    FILE_DATA_OK
+
+MU_DEFINE_ENUM(FILE_DATA_RESULT, FILE_DATA_RESULT_VALUES);
+
+typedef void *FILE_HANDLE;
+typedef void(*FILE_WRITE_CB)(void* userContext, bool isSuccess);
+typedef void(*FILE_READ_CB)(void* userContext, FILE_DATA_RESULT result, CONSTBUFFER_HANDLE content);
+
+MOCKABLE_FUNCTION_WITH_RETURNS(, FILE_HANDLE, file_open, const char*, fullFileName, int64_t, desire_file_size)(0, MU_FAILURE);

```

Comment:
>MOCKABLE_FUNCTION_WITH_RETURNS [](start = 0, length = 30)

Should be `MOCKABLE_FUNCTION`.

`MOCKABLE_FUNCTION_WITH_RETURNS` lets you specify the success and failure return values for mocking, which works well when it returns an enum or int (there are clear success/failure values). For a handle, the success value is variable so it's better to just leave this out.